diesel_codegen_new/src/ast_builder.rs

```diff
@@ -0,0 +1,16 @@
+use syn::{Ident, Ty, Path, PathSegment};
+
+pub fn ty_ident(ident: Ident) -> Ty {
```

Comment:
Do you think it makes sense to add these upstream as constructors? `syn::Ty::ident(i)`, `syn::Ty::path(p)`, `syn::Path::ident(i)` or something like that. And should the ident ones take `Ident` or `Into<Ident>` so they can be called with strings?